### PR TITLE
U4-9429 Make new create dialogs for code files (or rethink snippets)

### PR DESCRIPTION
**Pull request by emilwangaa**
This PR comes with some limitations:
- It's not possible to create folders - will be fixed in http://issues.umbraco.org/issue/U4-9481
- Tree menu logic is not fully implemented - will be fixed in http://issues.umbraco.org/issue/U4-9484
- It has a known bug in regards to creation where it doesn't load the editor properly - will be fixed in http://issues.umbraco.org/issue/U4-9483

### CODE REVIEW

File: src/Umbraco.Web.UI.Client/src/views/partialviewmacros/create.controller.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            navigationService.hideMenu();
+
+        }
+
+        function showCreateFolder() {
+            vm.creatingFolder = true;
+        }
+
+        function createFolder() {
+            alert("create folder");
```
Comment (warrenbuckley) [inline src/Umbraco.Web.UI.Client/src/views/partialviewmacros/create.controller.js]: I know this is coming in a future issue @madsrasmussen & @emilwangaa - but be good not to commit alert/console.log debugging stuff.

File: src/Umbraco.Web.UI.Client/src/views/partialviewmacros/create.controller.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+        function createPartialViewMacro(selectedSnippet) {
+
+            var snippet = null;
+
+            if(selectedSnippet && selectedSnippet.fileName) {
+                snippet = selectedSnippet.fileName;
+            }
+
+            $location.search('create', null);
```
Comment (warrenbuckley) [inline src/Umbraco.Web.UI.Client/src/views/partialviewmacros/create.controller.js]: @madsrasmussen I am not sure I understand this code here?
You are setting the create & snippet part of the URL to null , then changing the url/redirecting to the file editor & then re-adding back on the create & search snippet stuff.

File: src/Umbraco.Web.UI.Client/src/views/partialviewmacros/create.html [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            
+            <h5><localize key="create_createUnder">Create an item under</localize> {{currentNode.name}}</h5>
+
+            <!-- Main options -->
+            <div ng-if="!vm.showSnippets">
+                <ul class="umb-actions umb-actions-child">
+                    <li>
+                        <a href="" ng-click="vm.createPartialViewMacro()" umb-auto-focus>
+                            <i class="large icon-article"></i>
+                            <span class="menu-label">New Empty Partial View Macro</span>
```
Comment (warrenbuckley) [inline src/Umbraco.Web.UI.Client/src/views/partialviewmacros/create.html]: Can we localize this as well please @madsrasmussen

File: src/Umbraco.Web.UI.Client/src/views/partialviews/create.html [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            
+            <h5><localize key="create_createUnder">Create an item under</localize> {{currentNode.name}}</h5>
+
+            <!-- Main options -->
+            <div ng-if="!vm.showSnippets">
+                <ul class="umb-actions umb-actions-child">
+                    <li>
+                        <a href="" ng-click="vm.createPartialView()" umb-auto-focus>
+                            <i class="large icon-article"></i>
+                            <span class="menu-label">New Empty Partial View</span>
```
Comment (warrenbuckley) [inline src/Umbraco.Web.UI.Client/src/views/partialviews/create.html]: Same as before localize please :)

File: src/Umbraco.Web.UI.Client/src/views/scripts/create.html [hunk 5]
```diff
@@ -0,0 +1,47 @@
+<div class="umbracoDialog umb-dialog-body with-footer" ng-controller="Umbraco.Editors.Scripts.CreateController as vm" ng-cloak>
+
+    <div class="umb-pane" ng-if="!vm.creatingFolder">
+        <h5><localize key="create_createUnder">Create new partial view under</localize> {{currentNode.name}}</h5>
+
+        <ul class="umb-actions umb-actions-child">
+            <li>
+                <a href="" umb-auto-focus ng-click="vm.createFile()">
+                    <i class="large icon-script"></i>
+                    <span class="menu-label">Javascript file</span>
```
Comment (warrenbuckley) [inline src/Umbraco.Web.UI.Client/src/views/scripts/create.html]: Again localize please

File: src/Umbraco.Web.UI.Client/src/views/scripts/create.html [hunk 6]
```diff
@@ -0,0 +1,47 @@
+<div class="umbracoDialog umb-dialog-body with-footer" ng-controller="Umbraco.Editors.Scripts.CreateController as vm" ng-cloak>
+
+    <div class="umb-pane" ng-if="!vm.creatingFolder">
+        <h5><localize key="create_createUnder">Create new partial view under</localize> {{currentNode.name}}</h5>
```
Comment (warrenbuckley) [inline src/Umbraco.Web.UI.Client/src/views/scripts/create.html]: Copy/paste error the fallback message mentions partials and this is for JS

File: src/Umbraco.Web.UI.Client/src/views/partialviewmacros/create.controller.js (hunk 2 above)
Comment (madsrasmussen) [inline src/Umbraco.Web.UI.Client/src/views/partialviewmacros/create.controller.js]: The nulls are not needed - I have removed them

**Pull request merged**